api/src/main/java/org/apache/iceberg/types/PruneColumns.java

```diff
@@ -24,13 +24,24 @@
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types.ListType;
+import org.apache.iceberg.types.Types.MapType;
+import org.apache.iceberg.types.Types.StructType;
 
 class PruneColumns extends TypeUtil.SchemaVisitor<Type> {
   private final Set<Integer> selected;
+  private final boolean selectFullTypes;
 
-  PruneColumns(Set<Integer> selected) {
+  /**
+   * Visits a schema and returns only the fields selected by the id set.
+   * @param selected ids of elements to return
+   * @param selectFullTypes when true, selecting a nested type selects all subfields. When false selecting list or
+   *                        map types is undefined and forbidden.
```

Comment:
Minor: I normally opt to explain behavior in the method description and not in argument descriptions. Argument descriptions should state what the argument controls, but not all the implications. Otherwise, the argument descriptions are very long and reading the description doesn't cover the full behavior of the method. I'd probably change this to just "when true, selecting a nested type selects all subfields". And add a paragraph above that explains the behavior for structs, maps, and lists.